src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -252,12 +253,27 @@ internal struct NameDeclarationInfo
             {
                 result = IsFollowingTypeOrComma<VariableDeclarationSyntax>(token, semanticModel,
                      v => v.Type,
-                     v => v.Parent is LocalDeclarationStatementSyntax l ? l.Modifiers : default(SyntaxTokenList?),
+                     v => v.Parent is LocalDeclarationStatementSyntax l ? l.Modifiers :
+                          v.Parent is ForStatementSyntax ? default(SyntaxTokenList) : null as SyntaxTokenList?,
```

Comment:
The `as` is probably unnecessary.

Reply:
it is seriously weird to try to figure out ```default(SyntaxTokenList) : null as SyntaxTokenList?```.  Is this just:   ```v.Parent is ForStatementSyntax ? default(SyntaxTokenList) : default(SyntaxTokenList?)```?

Reply:
Yes it is. I like to be explicit about the `null` here. To me `? default(SyntaxTokenList) : default(SyntaxTokenList?)` seems weirder / harder to read.

Reply:
>We don't usually use default for null

We do in lots of code :)   Though it's not everywhere since it's a recent feature.  I don't mind null in general, but it seems strange that you are doing "null as SomeType?".   i.e. you're saying "i want the null value of this nullable type".  But htat's already easy to say with ```default(SomeType?)```.  That's already the easy way to say precisely that.

Reply:
could you do ```(SomeType?)null``` instead.  ```null as``` is def strange.  I could also find no non-test usages of that in our codebase.  Whereas we at least have ```(SomeType?)null``` in many places.

Reply:
I think this is more readable. It's easy to miss the question mark.

Reply:
sorry i must have sent the comment at the same time as you

Reply:
fixed in ad770dce343a57fa62c75368be125791cdc21ffb